### DISCUSSION TITLE
Livewire + spatie/laravel-data issue?

### DISCUSSION THREAD

**Discussion by robert-stanciu** [livewire/livewire#5579] [1. Help]
Hi community,

I recently started using spatie/laravel-data in my project and I enjoy it. I really want to benefit from its really powerful compatibility with Laravel.

So far, it really works OK if you are not updating `Data` properties using wire:model.
The dehydration and hydration work as expected and the metadata is correctly put in its place because all my `Data` classes are instances of `Wireable`.

The fun part comes when updating Carbon properties with wire:model

The Income class has a $date property.
```php
class Income extends Data implements Wireable
{
    use WireableData;

    public function __construct(
        #[WithCast(DateTimeInterfaceCast::class), WithTransformer(DateTimeInterfaceTransformer::class)]
        public Carbon $date = null
    ) {
    }
}

```

The IncomeForm component uses looks like this:
```php
class IncomeForm extends Component
{
    public ?Income $income = null;
}
```

The blade:
```html
<x-data wire:model="income.date"/>
```

When trying to update `income.data`, it throws this exception:
`Cannot assign string to property App\Libraries\Income::$from of type ?Carbon\Carbon`

Is there a way to force the `Data` object to recast the `$date` property?

**Comment by joshhanley** [comment]
@Dem0seQuence the issue is data coming from the front end is strings. 

So you will need to use an updating hook to validate the string and then convert it to a carbon instance. The trick is you need to use value by reference (noticed the & in the method signature).

```php
public function updatingIncomeDate(&$value)
{
    // Validate $value is a valid date string
    
    $value = Carbon::parse($value);
}
```

Hope this helps!
